internal/pkg/term/selector/selector.go

```diff
@@ -673,6 +693,36 @@ func filterWlsByName(wls []*config.Workload, wantedNames []string) []string {
 	return filtered
 }
 
+// Pipeline fetches all the pipelines in a workspace and prompts the user to select one.
+func (s *PipelineSelect) Pipeline(msg, help string) (*workspace.PipelineManifest, error) {
+	pipelines, err := s.ws.ListPipelines()
+	if err != nil {
+		return nil, err
+	}
+	if len(pipelines) == 0 {
+		return nil, errors.New("no pipelines found")
+	}
+	var pipelineNames []string
```

Comment:
noticed this pattern a few times. the `append` below might make multiple allocations despite we know the exact size from the start: it is exactly `len(pipelines)`, so in this and similar case I suggest to preallocate the slice: `pipelineNames := make([]string, len(pipelines))`.